unstructured/embed/mixedbreadai.py

```diff
@@ -0,0 +1,214 @@
+            additional_headers={"User-Agent": USER_AGENT},
+        )
+
+    @property
+    def num_of_dimensions(self):
+        """Get the number of dimensions for the embeddings."""
+        return np.shape(self.exemplary_embedding)
+
+    @property
+    def is_unit_vector(self) -> bool:
+        """Check if the embedding is a unit vector."""
+        return np.isclose(np.linalg.norm(self.exemplary_embedding), 1.0)
+
+    def _embed(self, texts: List[str]) -> List[List[float]]:
+        """
+        Embed a list of texts using the Mixedbread AI API.
+
+        Args:
+            texts (List[str]): List of texts to embed.
+
+        Returns:
+            List[List[float]]: List of embeddings.
+        """
+        batch_size = self.config.batch_size or len(texts)
+        batch_itr = range(0, len(texts), batch_size)
+
+        responses = []
+
+        for i in batch_itr:
+            batch = texts[i : i + batch_size]
+            response = self.client.embeddings(
+                model=self.config.model_name,
+                normalized=self.config.normalized,
+                encoding_format=self.config.encoding_format,
+                truncation_strategy=self.config.truncation_strategy,
+                dimensions=self.config.dimensions,
+                prompt=self.config.prompt,
+                request_options=self._request_options,
+                input=batch,
+            )
+            responses.append(response)
+        return [item.embedding for response in responses for item in response.data]
+
+    @staticmethod
+    def _add_embeddings_to_elements(elements, embeddings) -> List[Element]:
+        """
+        Add embeddings to elements.
+
+        Args:
+            elements (List[Element]): List of elements.
+            embeddings (List[List[float]]): List of embeddings.
+
+        Returns:
+            List[Element]: Elements with embeddings added.
+        """
+        assert len(elements) == len(embeddings)
+        elements_w_embedding = []
+        for i, element in enumerate(elements):
+            element.embeddings = embeddings[i]
+            elements_w_embedding.append(element)
+        return elements
+
+    def embed_documents(self, elements: List[Element]) -> List[Element]:
+        """
+        Embed a list of document elements.
+
+        Args:
+            elements (List[Element]): List of document elements.
+
+        Returns:
+            List[Element]: Elements with embeddings.
+        """
+        embeddings = self._embed([str(e) for e in elements])
+        return self._add_embeddings_to_elements(elements, embeddings)
+
+    def embed_query(self, query: str) -> List[float]:
+        """
+        Embed a query string.
+
+        Args:
+            query (str): Query string to embed.
+
+        Returns:
+            List[float]: Embedding of the query.
+        """
+        return self._embed([query])[0]
+
+    @EmbeddingEncoderConnectionError.wrap
+    @requires_dependencies(["mixedbread_ai"])
```

Comment:
please mimic voyageai embeddings here and add the `extras="embed-mixedbreadai"

https://github.com/Unstructured-IO/unstructured/blob/6b1d5f28bb9401b497f98d5dc0e780e51bcfeb38/unstructured/embed/voyageai.py#L72

And search the repo for `embed-voyageai` . This will allow a user to install the dependencies as `pip install "unstructured[embed-mixedbreadai]"` as per our other embedders and also connectors, etc

Reply:
fixed!